app/scripts/metamask-controller.js

```diff
@@ -1183,12 +1170,11 @@ module.exports = class MetamaskController extends EventEmitter {
    */
   setCurrentCurrency (currencyCode, cb) {
     try {
-      this.currencyController.setCurrentCurrency(currencyCode)
-      this.currencyController.updateConversionRate()
+      this.currencyRateController.currency = currencyCode
       const data = {
-        conversionRate: this.currencyController.getConversionRate(),
-        currentCurrency: this.currencyController.getCurrentCurrency(),
-        conversionDate: this.currencyController.getConversionDate(),
+        conversionRate: this.currencyRateController.state.conversionRate,
```

Comment:
Wow does that look cleaner!

Okay also I need [this line](https://github.com/bitpshr/gaba/blob/master/src/BaseController.ts#L122) explained to me.

Reply:
Great question, the `configure` function is a little dense. Whenever it's called with a new controller configuration object, each option passed is set on the controller itself as a member variable. For example, if I update a controller's configuration by calling `controller.configure({ foo: 'bar' })`, this will in turn set `controller.foo = 'bar'`. This is how controller configuration works: any option passed will always be set on the controller as a member, allowing controller authors to define custom setters in case anything special needs to happen when a particular configuration option is updated. See the TokenRatesController for an [example](https://github.com/bitpshr/gaba/blob/master/src/TokenRatesController.ts#L74) where a new polling interval is set up any time a new `interval` configuration option is set. I'll thoroughly cover this in the CONTRIBUTING documentation.

The specific line you linked to is what actually sets a config option on a controller instance:

```ts
(this as any)[key as string] = this.internalConfig[key];
```

...is the same as:

```js
this[key] = this.internalConfig[key];
```

...so that code is actually:

```js
// Store new internal config object
this.internalConfig = overwrite ? config : Object.assign(this.internalConfig, config);

// Iterate over each option
for (const key in this.internalConfig) {
    // If the option is actually defined...
    if (typeof this.internalConfig[key] !== 'undefined') {
        // Set it as a member on this instance, triggering any defined setters
        this[key] = this.internalConfig[key];
    }
}
```

The `as any` types are a way to appease the TypeScript compiler, which is necessary in this unique case. Since this is an abstract class, we can't define (or know) every possible config option on this base class. To get around this, we case `this` as an `any` type so TS basically ignores it.